packages/fxa-auth-server/lib/google-maps-services.ts

```diff
@@ -64,8 +64,23 @@ export class GoogleMapsService {
     if (status === Status.ZERO_RESULTS)
       throw new Error(`Could not find any results for address. (${address})`);
 
-    if (results.length > 1)
-      throw new Error(`Could not find unique results. (${address})`);
+    if (results.length > 1) {
+      const stateShortNames = results.map(
+        (result) =>
+          result.address_components.find((addressComponent) =>
+            addressComponent.types.includes(
+              PlaceType2.administrative_area_level_1
+            )
+          )?.short_name
+      );
```

Comment:
`Array.map` will guarantee that `stateShortNames` has the same length as `results`, so I don't check `if (results.length === stateShortNames.length) {...}`.

Also, it's possible a `result` may not have a state in any of its `addressComponents`, in which case the corresponding `stateShortNames` element will have a value of `undefined`.

Overall, for the 3 zipcodes I know return multiple `results` (10000, 42242 and 55868), none have every result with a state and every state matching, so all still yield error we were throwing before this patch. This means I have yet to find a valid zipcode where this change would make a difference.

Reply:
TIL a ZIP code can belong to multiple states! Neat!
https://stackoverflow.com/questions/12450960/find-zip-codes-associated-with-more-than-one-state/12451921
https://gis.stackexchange.com/questions/53918/determining-which-us-zipcodes-map-to-more-than-one-state-or-more-than-one-city

Reply:
> TIL a ZIP code can belong to multiple states! Neat!
> 

Yes, neat, in the same way that working with time zones is neat.  😛

Reply:
FWIW I did find an example zip (in Canada) that this patch would fix: J5L 0A3.